lib/queue.rb

```diff
@@ -1,30 +1,46 @@
 class Queue
-
+  
   def initialize
-    # @store = ...
-    raise NotImplementedError, "Not yet implemented"
+    @store = Array.new(10)
+    @front = @back = -1
   end
-
+  
   def enqueue(element)
-    raise NotImplementedError, "Not yet implemented"
+    if @front == -1 && @back == -1
+      @front = 0
+      @back = 1
+    end
+    if @front == @back
+      puts 'queue is full, sorry!'
+      return
+    end
+    @store[@back-1] = element
+    @back = (@back + 1) % @store.length # to get back to index 0
```

Comment:
you're having to do `@back-1` because you have set `@back` to 1 in the first if in this method and you wrote two separate ifs.  Converting `if @front == @back` into an elsif would make this a bit more straightforward.